jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -288,6 +293,19 @@ private static DiagnosticProperties runGC() {
 		return DiagnosticProperties.makeCommandSucceeded();
 	}
 
+	private static DiagnosticProperties getJstatClass(String diagnosticCommand) {
+		IPC.logMessage("jstat command : ", diagnosticCommand); //$NON-NLS-1$
+		StringWriter buffer = new StringWriter(100);
+		PrintWriter bufferPrinter = new PrintWriter(buffer);
```

Comment:
This should use the two-argument constructor (with `autoFlush` = true), or add an explicit call to `flush()` below.

Reply:
Usually `flush()` is required for disk file operation. In this case the `PrintWriter` wrappers a `StringWriter` which collects its output in a string buffer, its `flush()` is a no-op [1] hence there is no need for an explicit call.

[1] https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/660f897065982f667fb2bad1a9e66d8b8da496df/jdk/src/share/classes/java/io/StringWriter.java#L222-L226

Reply:
It might work that way now, but autoflush or an explicit call to `flush()` is still the right answer - you should not depend on implementation details (and it's the behavior of `PrintWriter` that's relevant here, not `StringWriter`).

Reply:
The method `getJstatClass` knows it is writing to a string buffer instead of a disk file, why add unnecessary bytecodes?
How about adding a comment that `flush()` is not required in this case?

Reply:
If there's any buffering in `PrintWriter`, flushing is necessary (auto or explicit).

Reply:
Updated PR and verified with a sanity build, it is ready for review.